Chat/Chat.cpp

```diff
@@ -0,0 +1,168 @@
+#include "Chat.h"
+#include "Network.h"
+#include <iostream>
+#include <thread>
+#include <chrono>
+
+#include <conio.h>
+
+using std::cout;
+using std::cin;
+using std::endl;
+
+
+Chat::Chat(){}
+
+
+Chat::~Chat()
+{
+	IOfflineMsg();
+}
+
+void Chat::IOnlineMsg()
+{
+	connected_network_->SendLogMsg(user_name_, LOG_ONLINE);
+	UserMsg msg = { msg_color_, user_name_, user_name_ + " is online!" };
+	SendMsg(msg);
+}
+
+void Chat::IOfflineMsg()
+{
+	connected_network_->SendLogMsg(user_name_, LOG_OFFLINE);
+	UserMsg msg = { msg_color_, user_name_, user_name_ + " left the chat!" };
+	SendMsg(msg);
+}
+
+void Chat::SendMsg(UserMsg msg)
+{
+	connected_network_->SendMsg(msg);
+	buffer_.clear();
+	AddMsg(msg);
+}
+
+int Chat::SendMsgTo(std::string name, UserMsg msg)
```

Comment:
I think, It would be better to pass parameters by reference, where it is possible.